<a href="https://colab.research.google.com/github/karishmasd/ML/blob/main/Heart.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Importing Libraries**

In [21]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

**Import Dataset**

In [22]:
from google.colab import drive
drive.mount('/content/drive/')

Drive already mounted at /content/drive/; to attempt to forcibly remount, call drive.mount("/content/drive/", force_remount=True).


In [23]:
df= pd.read_csv("/content/drive/MyDrive/Resume/Resume/Project no 1/heart.csv")

In [24]:
df

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0     63    1   3     145   233    1        0       150     0      2.3    0   
1     37    1   2     130   250    0        1       187     0      3.5    0   
2     41    0   1     130   204    0        0       172     0      1.4    2   
3     56    1   1     120   236    0        1       178     0      0.8    2   
4     57    0   0     120   354    0        1       163     1      0.6    2   
..   ...  ...  ..     ...   ...  ...      ...       ...   ...      ...  ...   
298   57    0   0     140   241    0        1       123     1      0.2    1   
299   45    1   3     110   264    0        1       132     0      1.2    1   
300   68    1   0     144   193    1        1       141     0      3.4    1   
301   57    1   0     130   131    0        1       115     1      1.2    1   
302   57    0   1     130   236    0        0       174     0      0.0    1   

     caa  thall  output  
0      0      1       1  
1      0      2       1  
2      0      2       1  
3      0      2       1  
4      0      2       1  
..   ...    ...     ...  
298    0      3       0  
299    0      3       0  
300    2      3       0  
301    1      3       0  
302    1      2       0  

[303 rows x 14 columns]

In [25]:
df= df.drop(['oldpeak','slp','thall'],axis=1)

In [19]:
df.head()

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  caa  output
0   63    1   3     145   233    1        0       150     0    0       1
1   37    1   2     130   250    0        1       187     0    0       1
2   41    0   1     130   204    0        0       172     0    0       1
3   56    1   1     120   236    0        1       178     0    0       1
4   57    0   0     120   354    0        1       163     1    0       1

**Data Analytics**

Age : Age of the patient


Sex : Sex of the patient


exang: exercise induced angina (1 = yes; 0 = no)



ca: number of major vessels (0-3)


cp : Chest Pain type chest pain type


Value 0: typical angina


Value 1: atypical angina


Value 2: non-anginal pain


Value 3: asymptomatic


trtbps : resting blood pressure (in mm Hg)


chol : cholestoral in mg/dl fetched via BMI sensor


fbs : (fasting blood sugar > 120 mg/dl) (1 = true; 0 = false)


rest_ecg : resting electrocardiographic results


Value 0: normal


Value 1: having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of > 0.05 mV)


Value 2: showing probable or definite left ventricular hypertrophy by Estes' criteria


thalach : maximum heart rate achieved


target : 0= less chance of heart attack 1= more chance of heart attack